lib/Mojo/DOM/CSS.pm

```diff
@@ -19,19 +19,52 @@ my $ATTR_RE   = qr/
 
 sub matches {
   my $tree = shift->tree;
-  return $tree->[0] ne 'tag' ? undef : _match(_compile(@_), $tree, $tree);
+  return $tree->[0] ne 'tag'
+    ? undef
+    : _match(_compile(@_), $tree, $tree, _root($tree));
 }
 
 sub select     { _select(0, shift->tree, _compile(@_)) }
 sub select_one { _select(1, shift->tree, _compile(@_)) }
 
+sub _absolute {
+  my ($selector) = @_;
+
+  # Traverse through selector to see if it's absolute.
+  # https://drafts.csswg.org/selectors-4/#absolutizing
+  for my $part (@$selector) {
+    next unless ref($part);
+
+    for my $pc (@$part) {
+      next unless ref($pc) && $pc->[0] eq 'pc';
+
+      # If ":root" or ":scope" are mentioned, it's absolute.
+      return 1 if $pc->[1] eq 'root' || $pc->[1] eq 'scope';
```

Comment:
I suspect this `:root` case is untested. At least there are no new tests for it.

Reply:
Any existing test involving `:root` would be affected by this. Since `:root` is already covered by tests, I thought those tests would suffice to check that functionality.